### ISSUE TITLE
Library Download Error

### ISSUE THREAD

**Issue by rajendrarox** [Sketchware-Pro/Sketchware-Pro#1359]
Sketchware Pro v6.4.0-SNAPSHOT-dc1cdea-minApi26 (150)
base.apk size: 138 MB (138158631 B)
Locale: en_IN
SDK version: 34
Brand: motorola
Manufacturer: motorola
Model: moto g84 5G

```
java.io.UncheckedIOException: java.io.IOException: close failed: ENOENT (No such file or directory)
	at java.util.zip.ZipFile$CleanableResource.run(ZipFile.java:858)
	at jdk.internal.ref.CleanerImpl$PhantomCleanableRef.performCleanup(CleanerImpl.java:212)
	at jdk.internal.ref.PhantomCleanable.clean(PhantomCleanable.java:133)
	at java.lang.Daemons$FinalizerDaemon.doClean(Daemons.java:388)
	at java.lang.Daemons$FinalizerDaemon.processReference(Daemons.java:358)
	at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:335)
	at java.lang.Daemons$Daemon.run(Daemons.java:131)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: java.io.IOException: close failed: ENOENT (No such file or directory)
	at libcore.io.IoBridge.closeAndSignalBlockedThreads(IoBridge.java:313)
	at java.io.RandomAccessFile.close(RandomAccessFile.java:726)
	at java.util.zip.ZipFile$Source.close(ZipFile.java:1460)
	at java.util.zip.ZipFile$Source.release(ZipFile.java:1407)
	at java.util.zip.ZipFile$CleanableResource.run(ZipFile.java:849)
	... 7 more
Caused by: android.system.ErrnoException: close failed: ENOENT (No such file or directory)
	at libcore.io.Linux.close(Native Method)
	at libcore.io.ForwardingOs.close(ForwardingOs.java:176)
	at libcore.io.BlockGuardOs.close(BlockGuardOs.java:113)
	at libcore.io.ForwardingOs.close(ForwardingOs.java:176)
	at libcore.io.IoBridge.closeAndSignalBlockedThreads(IoBridge.java:311)
	... 11 more

```

![Screenshot_20241106-151706](https://github.com/user-attachments/assets/5142cffd-6522-4b79-9e80-a16c9670e507)

Whenver I am Trying to download Library It crash 

Library: 'com.razorpay:checkout:1.6.20'

**Comment by ilyassesalama** [comment]
To be honest, the entire library downloader needs to be reworked. I just tried the library you mentioned and got many exceptions thrown.

**Comment by PranavPurwar** [comment]
I was able to recreate the crash. The library downloader is not failing. Its working alright. Its the [unzipping part](https://github.com/Sketchware-Pro/Sketchware-Pro/blob/cd67de27c65f6673ac7bbafaa4aab9810d87b3fb/app/src/main/java/mod/pranav/dependency/resolver/DependencyResolver.kt#L273)
 
MT manager can read the zip properly, but ZipFile implementation cannot. Likely an issue with the internal implementation. I did not want to add any other dependencies, but if you wish, you may use another library for unzipping.

**Comment by PranavPurwar** [comment]
> To be honest, the entire library downloader needs to be reworked. I just tried the library you mentioned and got many exceptions thrown.

I did not find any other exceptions to be thrown, can you send more details about them?

**Comment by ilyassesalama** [comment]
> > To be honest, the entire library downloader needs to be reworked. I just tried the library you mentioned and got many exceptions thrown.
> 
> I did not find any other exceptions to be thrown, can you send more details about them?

The thrown exceptions are the same as the one you mentioned. In the unzipping step.

**Comment by rajendrarox** [comment]
Now fixed checkout #1362
